go_modules

```diff
@@ -176,13 +197,15 @@ def main():
         log.info(f"Vendor go.mod dependencies to {vendor_tarname}")
         vendor_tarfile = os.path.join(outdir, vendor_tarname)
         vendor_dir = os.path.join(go_mod_dir, "vendor")
-        with tarfile.open(vendor_tarfile, "w:" + args.compression) as tar:
+        tarmode = "w:"
+        if args.compression != "tar":
+            tarmode += args.compression or DEFAULT_COMPRESSION
+        with tarfile.open(vendor_tarfile, tarmode) as tar:
             tar.add(vendor_dir, arcname=("vendor"))
 
         # remove extracted Go application source
         try:
-            to_remove = os.path.join(outdir, basename)
-            shutil.rmtree(to_remove)
+            shutil.rmtree(tempdir)
```

Comment:
With the `with` you can remove all this `try` block.

Reply:
Sure, but I cannot figure out how to handle the error when the directory cannot be removed for some reason - as this exception would happen in a cleanup of the context manager.
@aplanas Any hint?

Reply:
Reading the doc about [TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) we can see that it will remove automatically the directory for you. If it fail, you will fail too.

Also inspecting the [code](https://github.com/python/cpython/blob/3.10/Lib/tempfile.py#L827) we can see that the clean method goes much much farther that the `rmtree` in your code, like adjusting permissions, behaving differently when an exception happens, etc. So IMHO just use the context manager with TemporaryDirectory and it will be cleaned automatically.